lib/setupHooks/installFromCargoBuildLogHook.sh

```diff
@@ -13,12 +13,18 @@ function installFromCargoBuildLog() (
   logs=$(jq -R 'fromjson?' <"${log}")
 
   local select_non_test='select(.reason == "compiler-artifact" and .profile.test == false)'
-  local select_bins="${select_non_test} | .executable | select(.!= null)"
+  local select_non_dep='select((contains("/release/deps/") or contains("/debug/deps/"))| not)'
+  local select_bins="${select_non_test}"'| .executable | select(.!= null) | '"${select_non_dep}"
   local select_lib_files="${select_non_test}"'
-    | select(.target.kind | contains(["staticlib"]) or contains(["cdylib"]))
+    | select(.target.kind
+        | contains(["cdylib"])
+        or contains(["dylib"])
+        or contains(["lib"])
```

Comment:
Question: why do we want to install `lib` by default? According to [the reference](https://doc.rust-lang.org/reference/linkage.html) `lib` means 
> This is an ambiguous concept as to what exactly is produced because a library can manifest itself in several forms. The purpose of this generic lib option is to generate the "compiler recommended" style of library. The output library will always be usable by rustc, but the actual type of library may change from time-to-time.

I'm not entirely sure what this means or how it would be usable (but if it's something that can be used by other builds/linkers I don't mind including it; I'd just like to see an example to understand things better!)

Reply:
I'd guess for some niche, very low-level use cases where the consumers of `crane` would want to e.g. build custom `cargoArtifacts`.
Happy to remove it

Reply:
Yeah I'd like to keep things as simple as possible for now, if someone has a use case which needs it we can revisit